cmd/arena/commands/heapster.go

```diff
@@ -0,0 +1,117 @@
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/bitly/go-simplejson"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+const (
+	DefaultHeapsterScheme      = "http"
+	DefaultHeapsterServiceName = "heapster"
+	DefaultHeapsterPort        = "80"
+	HeapsterApiPath            = "/api/v1/model"
+)
+
+var httpClient = &http.Client{Timeout: 30 * time.Second}
+
+func getHeapsterEndpointUrl(client *kubernetes.Clientset) (string, error) {
+	heapsterService, err := client.CoreV1().Services(metav1.NamespaceSystem).Get(DefaultHeapsterServiceName, metav1.GetOptions{})
+	if err != nil {
+		return "", err
+	}
+	heapsterEndpointUrl := DefaultHeapsterScheme + "://" + heapsterService.Spec.ClusterIP + ":" + DefaultHeapsterPort + HeapsterApiPath
```

Comment:
`heapsterService.Spec.ClusterIP` Should be ApiServer IP .
If arena execute outside cluster nodes, it can't reach the heapster ClusterIP.

Reply:
Yes, you are right, we should consider this.